Reorganize "meta" fields

pivotaljohn opened the pull request:
- remove "yaml:" struct field tags; our YAML is being processed through
  sigs.k8s.io/yaml (which performs (un)marshaling through the JSON
  parser, not directly through go-yaml/yaml.
- lowercase all field names (consistent with YAML around it, making it easier to read)
- promote "Type" fields as top-level keys for each type of meta.

jtigger committed:
Lowercase "meta" fields

- remove "yaml:" struct field tags; our YAML is being processed through
  sigs.k8s.io/yaml (which performs (un)marshaling through the JSON
  parser, not directly through go-yaml/yaml.

cppforlife commented on pkg/kbld/config/image_meta.go:
```diff
@@ -26,33 +26,33 @@ const (
-	Dirty     bool
-	Tags      []string `json:",omitempty" yaml:",omitempty"`
+	Type      string   `json:"type"` // always set to GitMeta
+	RemoteURL string   `json:"remoteUrl"`
```
here URL should still be upper case.

pivotaljohn replied:
done.

cppforlife commented:
since we are changing field names... should we just go ahead and change to something like this:

```
metas:
- git:
    remoteURL ...
- git:
    remoteURL: ...
- preresolved:
    ...
```

pivotaljohn commented:
> since we are changing field names... should we just go ahead and change to something like this:
> 
> ```
> metas:
> - git:
>     remoteURL ...
> - git:
>     remoteURL: ...
> - preresolved:
>     ...
> ```

❤️  — promoting "type:" as a top-level key fixes a complaint I have about it being difficult to visually parse this YAML given that the processing (through K8s' YAML parser marshaling/unmarshaling to/from the JSON parser) sorts the keys alphabetically.

We're breaking this "API", let's do it up right. 👍🏻

jtigger committed:
Name type-of-meta as a top-level key

... to make the meta section easier to read (by making more clear what
    type each entry in the meta).

Also, fixes `remoteUrl` ==> `remoteURL`

Commit:
simplify parsing of metas

Commit:
compile tests as part of ./hack/build.sh

cppforlife commented:
superseded by #169
